Destination/Destination.php

```diff
@@ -3,15 +3,25 @@
 namespace Zenstruck\BackupBundle\Destination;
 
 use Psr\Log\LoggerInterface;
+use \Countable;
+use \IteratorAggregate;
 
 /**
  * @author Kevin Bond <kevinbond@gmail.com>
  */
-interface Destination
+interface Destination extends Countable, IteratorAggregate
```

Comment:
It is not only about GUI -> if destination can not be iteratable/readable, than it is a black hole that swallows backups only -> we can not list files, see what is in it, what is the amount of occupied space… etc…. 

In regards to S3Cmc -> according to the documentation, it is possible to navigate trough buckets, to list files, get file infos, etc…, see: http://s3tools.org/usage <http://s3tools.org/usage>

It would be quite unusual that it is impossible to get details about the storage…

GUI should be separate bundle, I do not mind that -> but I really believe that Destination should be iteratable and should allow to the developer to list trough all current stored backups. 

What is the issue having Destination implementing possibility to iterate trough current backups - what are the risks, side effects, dangers?


> On Sep 11, 2015, at 2:49 PM, Kevin Bond <notifications@github.com> wrote:
> 
> In Destination/Destination.php <https://github.com/kbond/ZenstruckBackupBundle/pull/3#discussion_r39267362>:
> 
> >  
> >  /**
> >   * @author Kevin Bond <kevinbond@gmail.com>
> >   */
> > -interface Destination
> > +interface Destination extends Countable, IteratorAggregate
> Ok, I see what you mean.
> 
> If you are building a GUI, do you not want the functionality to remove backups as well? We should think about what we want the GUI to do. That functionality might be out the scope of the Destination. I intended the destination to only push a file.
> 
> For instance, the S3CmdDestination can really never do more than push - it can't list or remove (well i suppose it could but it would be exceedingly difficult).
> 
> I think we need a separate system for "administrating" the backups. Maybe even a separate bundle.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/kbond/ZenstruckBackupBundle/pull/3/files#r39267362>.
>